include/swift/ABI/Metadata.h

```diff
@@ -116,6 +116,12 @@ struct MetadataDependency {
   }
 };
 
+/// Prefix of a metadata header, containing a pointer to the
+/// type layout string.
+struct TargetTypeMetadataLayoutPrefix {
+  const uint8_t *layoutString;
```

Comment:
We should template this on `Runtime` like the others. It might be useful to be able to read these strings from out of process, and it would be good to have the correct layout when `Runtime` doesn't match the native environment This should probably also use a signed pointer (see `TargetSignedPointer`) since corrupting this string could be used to manipulate/smash other bits of memory.